### PR TITLE
add LoadedBridgeCliConfig

### PR DESCRIPTION
**Pull request by longbowlu**
## Description 

Describe the changes or additions included in this PR.

## Test plan 

How did you test the new or updated feature?

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:

### CODE REVIEW

File: crates/sui-bridge/src/config.rs [hunk 1]
```diff
@@ -417,6 +418,43 @@ pub fn read_bridge_client_key(path: &PathBuf) -> Result<SuiKeyPair, anyhow::Erro
...
+pub fn read_key(path: &PathBuf, require_secp256k1: bool) -> Result<SuiKeyPair, anyhow::Error> {
+    if !path.exists() {
+        return Err(anyhow::anyhow!("Key file not found at path: {:?}", path));
+    }
+    let file_contents = std::fs::read_to_string(path)?;
+    let contents = file_contents.as_str().trim();
+
+    // Try base64 encoded SuiKeyPair `flag || privkey`
+    if let Ok(key) = SuiKeyPair::decode_base64(contents) {
+        if require_secp256k1 && !matches!(key, SuiKeyPair::Secp256k1(_)) {
```
Comment (dariorussi) [inline crates/sui-bridge/src/config.rs:435]: I guess you could use the `if let` pattern for a single match, if you found it more expressive

File: crates/sui-bridge/src/config.rs [hunk 2]
```diff
@@ -417,6 +418,43 @@ pub fn read_bridge_client_key(path: &PathBuf) -> Result<SuiKeyPair, anyhow::Erro
         .map_err(|e| anyhow!("Error decoding authority key: {:?}", e))
 }
 
+/// Read a SuiKeyPair from a file. The content could be any of the following:
+/// - Base64 encoded `flag || privkey` for ECDSA key
+/// - Base64 encoded `privkey` for Raw key
+/// - Hex encoded `privkey` for Raw key
+/// If `require_secp256k1` is true, it will return an error if the key is not Secp256k1.
+pub fn read_key(path: &PathBuf, require_secp256k1: bool) -> Result<SuiKeyPair, anyhow::Error> {
```
Comment (dariorussi) [inline crates/sui-bridge/src/config.rs:426]: this model of trying on or the other feels a bit fragile. 
Is there no other way to do it?

File: crates/sui-bridge/src/tools/mod.rs [hunk 3]
```diff
@@ -245,33 +249,111 @@ pub struct BridgeCliConfig {
...
+        let sui_bridge = EthSuiBridge::new(cli_config.eth_bridge_proxy_address, provider.clone());
+        let eth_bridge_committee_proxy_address: EthAddress = sui_bridge.committee().call().await?;
+        let eth_bridge_limiter_proxy_address: EthAddress = sui_bridge.limiter().call().await?;
+
+        let eth_address = eth_signer.address();
+        let eth_chain_id = provider.get_chainid().await?;
+        let sui_address = SuiAddress::from(&sui_key.public());
+        println!("Using Sui address: {:?}", sui_address);
+        println!("Using Eth address: {:?}", eth_address);
+        println!("Using Eth chain: {:?}", eth_chain_id);
```
Comment (dariorussi) [inline crates/sui-bridge/src/tools/mod.rs:333]: is `println` the way we print something in the tool? or is this something that should be removed or done differently?

**Review by dariorussi — APPROVED** [review]

**Comment by longbowlu** [comment]
https://github.com/MystenLabs/sui/pull/17729

**Pull request closed without merging**